fix immutable/const types & add type convenience

WebFreak001 opened the pull request:
Before it was not properly possible to use
```
static immutable name = interfaceName(...)
```
in a message, which is now fixed

This probably requires a minor bump as there _might_ be potential breakage, especially when APIs before did not take immutable/const into account and got some undefined behavior. Please test with your applications!

Works fine for me with my main application using dbus and the tests pass too

thaven commented on source/ddbus/thin.d:
```diff
@@ -696,6 +701,21 @@ struct DBusAny {
+  /// Returns: true if this variant is an integer or a floating point value.
+  bool typeIsNumeric() const @property {
+    return typeIsIntegral || typeIsFloating;
+  }
```
I think these are more at home in utils.d

WebFreak001 replied:
Hm I would agree with that but utils isn't imported with `import ddbus;` so using it would require a second import and it feels like you are importing internal utilities there, even though these types are meant for regular usage and probably will only be used with these variants.

thaven reviewed (commented):
I don't have time for a thorough review now, but at a first glance most of it looks good.

ˋUnqual!Tˋ would also strip ˋsharedˋ away from T, which might cause undefined behavior, so we might need some explicit checks to disallow ˋsharedˋ types.

WebFreak001 committed:
fix immutable/const types & add type convenience

Before it was not properly possible to use
static immutable name = interfaceName(...)
in a message, which is now fixed

WebFreak001 commented:
rebased master, moved type functions into utils.d (and kept utility wrappers inside DBusAny for discovery)

WebFreak001 reviewed (commented):
as all the Unqual casts are only on primitive types or immutable types (string, ObjectPath), casting away shared is of no concern.

Therefore I'm going to merge the current state.
